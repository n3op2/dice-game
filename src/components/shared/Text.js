import React from 'react';

const Text = ({ children = [] }) => {
    const isValidChildren =
        typeof children === 'string' ||
        children.every((child) => typeof child === 'string');

    if (!isValidChildren) throw new Error('invalid children');
    return <div className="text">{children}</div>;
};

export default Text;
