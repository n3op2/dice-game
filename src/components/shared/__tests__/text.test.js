import React from 'react';
import { shallow } from 'enzyme';
import Text from '../Text';

const Component = <Text>some random text here</Text>;

describe('Text Component', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(Component);
    });

    describe('if not all children are typeof string', () => {
        test('throws an error and does not render', () => {
            expect(() =>
                shallow(
                    <Text>
                        text{[{}, {}, {}]}
                        {class a {}}
                    </Text>
                )
            ).toThrowError('invalid children');
        });
    });

    test('it renders component', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
