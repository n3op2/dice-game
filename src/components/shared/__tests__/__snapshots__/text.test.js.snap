// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component it renders component 1`] = `
<div
  className="text"
>
  some random text here
</div>
`;
